godotproject/Player.gd

```diff
@@ -92,17 +104,110 @@ func get_input():
+	#     => frog experiences constant acceleration (air_horiz_accel) in input direction until they
+	#        reach their max speed (air_speed_limit), beyond which the speed doesn't increase
+	# 
+	#   - user is inputting direction OPPOSITE the dir'n the frog is going:
+	#     => frog experiences boosted constant acceleration (air_horiz_accel * air_turn_coefficient)
+	#        in the inputted direction (still obeying the maximum speed limit)
+	# 
+	# When a direction is not input while in the air, momentum is kept the same since that logically
+	# makes sense and seems to play OK.
+	# 
+	# There is a horizontal speed cap while in the air, but this only affects the user's inputs.
+	# This means they can't exceed this speed cap by inputting left/right, but when they eject
+	# out of a swing, they will be allowed to be above the speed cap. If they push in the dir'n
+	# they are going, then they won't lose speed. If they push in the dir'n opposite their
+	# velocity, then they will slow down using the same "air_turn_coefficient" as normal.
+	# 
+	# This conservation of momentum above the speed cap basically allows the user to keep their
+	# momentum for as long as they are in the air, slowing down when they touch ground.
+	# They can keep the full momentum if they can jump as soon as they touch the ground.
+	# 
+	###############################################################################################
+	
+	if on_floor_last_frame:
+		# Handle ground movement
+		if is_zero(user_direction.x):
+			# If the user isn't inputting a direction, slow them down toward stopped
+			var vs = sign(velocity.x)
+			velocity.x += -vs * walk_horiz_accel * delta * friction_coeff
+			if sign(velocity.x) * vs < 0:
+				# User is going in opposite dir'n then they were originally.
+				# Make them come to a stop.
+				velocity.x = 0
+		else:
+			# Make the user move in the direction they're pointing if they're not already
+			# at the speed cap
+			if sign(user_direction.x) * velocity.x < walk_speed_limit:
+				var turning_bonus = 1
+				# Add additional speed if the user is turning around
+				# (dir'n of user_direction.x doesn't match dir'n of velocity.x)
+				if sign(user_direction.x) * sign(velocity.x) < 0:
+					turning_bonus = ground_turn_coefficient
+				velocity.x += user_direction.x * walk_horiz_accel * delta * turning_bonus
+				# Limit horiz. speed to speed cap
+				velocity.x = sign(velocity.x) * min(abs(velocity.x), walk_speed_limit)
+			else:
+				# If they are at or above the speed cap, gradually slow them down towards
+				# speed cap (to stop them from going too fast)
+				velocity.x -= user_direction.x * going_too_fast_slowdown_accel * delta
+				# Don't decrease speed past the speed cap!
+				velocity.x = sign(velocity.x) * max(abs(velocity.x), walk_speed_limit)
+	else:
+		# Handle air movement
+		if is_zero(user_direction.x):
+			# If the user isn't inputting a direction, don't adjust their velocity
+			pass
```

Comment:
can we invert the if logic so it doesnt have the pass and wont need an else

Reply:
ie 
if not is_zero(...): 
    # the else statement

Reply:
yeye, I'll do now and do another commit and push